File: commons-aws/src/test/java/test/pl/chilldev/commons/aws/sns/NotificationHandlerTest.java
/**
 * This file is part of the ChillDev-Commons.
 *
 * @license http://mit-license.org/ The MIT license
 * @copyright 2017 - 2018 © by Rafał Wrzeszcz - Wrzasq.pl.
 */

package test.pl.chilldev.commons.aws.sns;

import java.util.Arrays;
import java.util.function.Consumer;

import com.amazonaws.services.lambda.runtime.events.SNSEvent;
import org.junit.Rule;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.Mockito;
import org.mockito.junit.MockitoJUnit;
import org.mockito.junit.MockitoRule;
import pl.chilldev.commons.aws.sns.NotificationHandler;

public class NotificationHandlerTest
{
    @Rule
    public MockitoRule mockito = MockitoJUnit.rule();

    @Mock
    private Consumer<SNSEvent.SNS> messageHandler;

    @Test
    public void process()
    {
        SNSEvent.SNS message1 = new SNSEvent.SNS();
        message1.setMessage("msg1");
        SNSEvent.SNSRecord record1 = new SNSEvent.SNSRecord();
        record1.setSns(message1);
        SNSEvent.SNS message2 = new SNSEvent.SNS();
        message2.setMessage("msg2");
        SNSEvent.SNSRecord record2 = new SNSEvent.SNSRecord();
        record2.setSns(message2);

        SNSEvent event = new SNSEvent();
        event.setRecords(Arrays.asList(record1, record2));

        NotificationHandler handler = new NotificationHandler(this.messageHandler);
        handler.process(event);

        Mockito.verify(this.messageHandler).accept(message1);
        Mockito.verify(this.messageHandler).accept(message2);
    }
}

File: commons-aws/src/test/kotlin/test/pl/wrzasq/commons/aws/messaging/sns/NotificationHandlerTest.kt
/*
 * This file is part of the pl.wrzasq.commons.
 *
 * @license http://mit-license.org/ The MIT license
 * @copyright 2017 - 2019, 2021 © by Rafał Wrzeszcz - Wrzasq.pl.
 */

package test.pl.wrzasq.commons.aws.messaging.sns

import com.amazonaws.services.lambda.runtime.events.SNSEvent
import com.amazonaws.services.lambda.runtime.events.SNSEvent.SNS
import com.amazonaws.services.lambda.runtime.events.SNSEvent.SNSRecord
import io.mockk.every
import io.mockk.impl.annotations.MockK
import io.mockk.junit5.MockKExtension
import io.mockk.just
import io.mockk.runs
import io.mockk.verify
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import pl.wrzasq.commons.aws.messaging.sns.NotificationHandler

@ExtendWith(MockKExtension::class)
class NotificationHandlerTest {
    @MockK
    lateinit var messageHandler: (SNS) -> Unit

    @Test
    fun process() {
        every { messageHandler(any()) } just runs

        val message1 = SNS()
        message1.message = "msg1"

        val record1 = SNSRecord()
        record1.setSns(message1)

        val message2 = SNS()
        message2.message = "msg2"

        val record2 = SNSRecord()
        record2.setSns(message2)

        val event = SNSEvent()
        event.records = listOf(record1, record2)

        val handler = NotificationHandler(messageHandler)
        handler.process(event)

        verify { messageHandler(message1) }
        verify { messageHandler(message2) }
    }
}
